Source/EasyNetQ/Producer/DefaultPubSub.cs

```diff
@@ -37,31 +37,32 @@ IAdvancedBus advancedBus
             this.advancedBus = advancedBus;
         }
 
+        /// <inheritdoc />
         public virtual async Task PublishAsync<T>(T message, Action<IPublishConfiguration> configure, CancellationToken cancellationToken)
         {
             Preconditions.CheckNotNull(message, "message");
             Preconditions.CheckNotNull(configure, "configure");
 
+            var cts = CreateCancellationTokenSource(cancellationToken);
```

Comment:
I don't understand why you introduced this variable to just pass token in declare/publish.